### PR TITLE
Use goreleaser to build and package summon-conjur

### PR DESCRIPTION
**Pull request by ghost**
This pull request changes how we build and package summon-conjur. I've implemented [goreleaser](https://github.com/goreleaser/goreleaser), a tool that lets you use a `.goreleaser.yml` file to specify how to build and package your go app. 

Using goreleaser comes with some nice benefits, like:
- consistent build and package environment, declarative yaml configuration
- builds deb/rpm packages
- creates homebrew file, ready to upload to https://github.com/cyberark/homebrew-tools

goreleaser can also automatically create github releases and upload artifacts to them, but I opted out of that in the config file for this project. I think having the homebrew file in CI artifacts makes it easy to add to homebrew-tools. Having the tarballs and shasums makes it easy to upload to github releases.

See the artifacts here:
- https://jenkins.conjur.net/job/cyberark--summon-conjur/job/use-goreleaser/
- https://gitlab.com/cyberark/summon-conjur/-/jobs/86779196/artifacts/browse/dist/

See a full list of options here: https://goreleaser.com/customization/

This is the first test of goreleaser. I think this was a good experience and we should consider adopting this tool for other go projects. We could build and push a image to dockerhub so multiple projects can use it easily.

I will follow this merge up with adding summon-conjur to homebrew repo.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -1,36 +1,9 @@
...
-  echo "....."
-
-  docker-compose run --rm \
-    -e GOOS=$GOOS -e GOARCH=$GOARCH \
-    builder \
-      build -v -o $OUTPUT_DIR/summon-conjur-$GOOS-$GOARCH
-done
+docker-compose build --pull goreleaser
+docker-compose run --rm -T --entrypoint sh goreleaser -es <<EOF
+dep ensure --vendor-only
```
Comment (sgnn7) [inline build.sh:7]: Pinged on slack but why do we dep ensure outside of the dockerfile? We would lose caching this way if I'm not mistaken.

**Review by sgnn7 — CHANGES_REQUESTED** [review]
@dustinmm80 Just one comment but looks reasonable

**Review by sgnn7 — APPROVED** [review]

**Comment by sgnn7** [comment]
CR pass (with pending task of maybe caching the dep dependencies to be opened)

File: build.sh (hunk 1 above)
Comment (ghost) [reply build.sh:7]: From Slack discussion

---

i think you’re probably right, but that we should defer caching for now.

as you say, releases not very frequent and also these builds are running on distributed CI so whether a cache is available on the node or not is up to the scheduler

i can open caching as an issue when this image has a standalone src repo.

**Pull request merged**